app/src/main/java/com/eveningoutpost/dexdrip/Models/LibreBlock.java

```diff
@@ -129,29 +139,57 @@ public static LibreBlock getLatestForTrend() {
         return getLatestForTrend(JoH.tsl() - Constants.DAY_IN_MS, JoH.tsl() );
     }
 
+    static LibreBlock getFromCursor(Cursor cursor) {
+        LibreBlock libreBlock  = new LibreBlock();
+
+        libreBlock.timestamp = cursor.getLong(cursor.getColumnIndex("timestamp"));
+        libreBlock.byte_start = cursor.getLong(cursor.getColumnIndex("bytestart"));
+        libreBlock.byte_end = cursor.getInt(cursor.getColumnIndex("byteend"));
+        libreBlock.reference = cursor.getString(cursor.getColumnIndex("reference"));
+        libreBlock.blockbytes = cursor.getBlob(cursor.getColumnIndex("blockbytes"));
+        libreBlock.calculated_bg = cursor.getDouble(cursor.getColumnIndex("calculatedbg"));
+        libreBlock.uuid = cursor.getString(cursor.getColumnIndex("uuid"));
+        libreBlock.patchUid = cursor.getBlob(cursor.getColumnIndex("patchUid"));
+
+        libreBlock.patchInfo = cursor.getBlob(cursor.getColumnIndex("patchInfo"));
+        return libreBlock;
+    }
 
     public static LibreBlock getLatestForTrend(long start_time, long end_time) {
 
-        return new Select()
-                .from(LibreBlock.class)
-                .where("bytestart == 0")
-                .where("byteend >= 344")
-                .where("timestamp >= ?", start_time)
-                .where("timestamp <= ?", end_time)
-                .orderBy("timestamp desc")
-                .executeSingle();
+        SQLiteDatabase db = Cache.openDatabase();
+        // Using this syntex since there is no way to tell the DB which index to use.
```

Comment:
The following change has made the query much faster.
It obviously does not mean that things can not be made better.
Changing the indexes is possible but:
1) very problematic to make sure everyone will get the correct index. (In the past changing indexes / removing fields made a lot of problems to users) For example after upgrade or downgrade of the version.
2) Will require more place in the DB. The current indexes works very well.

Reply:
> Will require more place in the DB. The current indexes works very well.

Really? The tables are over-indexed and the indices cannot be used because they are not composite. But I agree that we should use your workaround here and fix it in general with a different PR.